File: backend/app/models/busqueda.ts
export interface BusquedaProceso {
  buscar?: string;
  estado?: string;
  tipo_proceso?: string;

  fecha_publicacion_desde?: string;
  fecha_publicacion_hasta?: string;

  fecha_presentacion_desde?: string;
  fecha_presentacion_hasta?: string;

  limit?: number;
}

File: backend/app/models/busqueda.py
from datetime import date
from pydantic import BaseModel


class BusquedaProceso(BaseModel):
    buscar: str | None = None

    estado: str | None = None
    tipo_proceso: str | None = None
    tipo_contrato: str | None = None

    fecha_publicacion_desde: date | None = None
    fecha_publicacion_hasta: date | None = None

    limit: int = 50
